mrcfile/mrcobject.py

```diff
@@ -157,58 +159,100 @@ def _create_default_header(self):
                                                     time)
         header.nlabl = 1
-        
+
         self.reset_header_stats()
-    
+
     @property
     def header(self):
         """Get the header as a :class:`numpy record array <numpy.recarray>`."""
         return self._header
-    
+
     @property
     def extended_header(self):
         """Get the extended header as a :class:`numpy array <numpy.ndarray>`.
-        
-        If this :class:`MrcObject` was read from a file and the extended header
-        type was recognised, its dtype will be set appropriately. (Currently
-        the only supported types are ``'FEI1'`` and ``'FEI2'``.) Otherwise, the
-        dtype will be void (raw data, dtype ``'V'``). If the actual data type
+
+        The dtype will be void (raw data, dtype ``'V'``). If the actual data type
         of the extended header is known, the dtype of the array can be changed
-        to match.
-        
+        to match. For supported types (e.g. ``'FEI1'`` and ``'FEI2'`), the
+        indexed part of the extended header (excluding any zero padding) can be
+        accessed using :meth:`indexed_extended_header`
+
         The extended header may be modified in place. To replace it completely,
         call :meth:`set_extended_header`.
         """
         return self._extended_header
-    
+
+    @property
+    def indexed_extended_header(self):
+        """Get the indexed part of the extended header as a
+        :class:`numpy array <numpy.ndarray>` with the appropriate dtype set.
+        Currently only ``'FEI1'`` and ``'FEI2'` extended headers are supported.
+        Modifications to the indexed extended header will not change the
+        extended header data recorded in this :class:`MrcObject`. If the
+        extended header type is unrecognised or extended header data is not of
+        sufficient length a warning will be produced and the indexed extended
+        header will be None."""
+
+        # Use the header's byte order for the extended header
+        dtype = get_ext_header_dtype(self.header.exttyp,
+                                     self.header.mode.dtype.byteorder)
+
+        # Interpret one element
+        try:
+            if self._extended_header.size < dtype.itemsize:
+                raise ValueError
+            first = self._extended_header[0:dtype.itemsize]
+            first.dtype = dtype
+            if first["Metadata size"][0] != dtype.itemsize:
+                raise ValueError
+        except ValueError:
+            warnings.warn("The header has exttyp '{}' but the extended header "
+                            "cannot be interpreted as that type"
+                            .format(self.header.exttyp), RuntimeWarning)
+            return None
+
+        nbytes = int(self.header["nz"]) * dtype.itemsize
+        try:
+            if self._extended_header.size < nbytes:
+                raise ValueError
+            full = self._extended_header[0:nbytes]
```

Comment:
```suggestion
            if self.extended_header.size < dtype.itemsize:
                raise ValueError
            first = self.extended_header[0:dtype.itemsize]
            first.dtype = dtype
            if first["Metadata size"][0] != dtype.itemsize:
                raise ValueError
        except ValueError:
            warnings.warn("The header has exttyp '{}' but the extended header "
                            "cannot be interpreted as that type"
                            .format(self.header.exttyp), RuntimeWarning)
            return None

        nbytes = int(self.header["nz"]) * dtype.itemsize
        try:
            if self.extended_header.size < nbytes:
                raise ValueError
            full = self.extended_header[0:nbytes]
```

I think we may as well use the public `extended_header` attribute here, it should be equivalent and makes the code slightly easier to read